fieldservice_route_availability/models/fsm_order.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2025 APSL-Nagarro Antoni Marroig
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class FSMRoute(models.Model):
+    _inherit = "fsm.order"
+
+    @api.constrains("scheduled_date_start", "location_id")
+    def check_black_out_days(self):
+        for order in self:
+            if order.fsm_route_id and order.scheduled_date_start:
+                for group in order.fsm_route_id.fsm_blackout_group_ids:
+                    match = group.fsm_blackout_day_ids.filtered(
+                        lambda x, order=order: (
+                            (not x.zip and x.date == order.scheduled_date_start.date())
+                            or (
+                                x.zip
+                                and x.date == order.scheduled_date_start.date()
+                                and x.zip == order.zip
+                            )
+                        )
+                    )
+                    if match:
+                        raise ValidationError(
+                            _(
+                                "The date %(date)s is a blackout day for field"
+                                " service operations on this route."
+                            )
+                            % {
+                                "date": order.scheduled_date_start.date().strftime(
+                                    "%d/%m/%Y"
+                                )
+                            }
```

Comment:
```suggestion
                            self.env._(
                                "The date %(date)s is a blackout day for field"
                                " service operations on this route.",
                                date=format_date(order.scheduled_date_start),
                            )
```

> [!NOTE]
> `format_date` is imported from `odoo.tools.misc`


A bunch of things in this suggestion:
1. Use `self.env._` instead of `_` for more efficient gettext translation
2. Use gettext's string interpolation directly, which lets Odoo recover gracefully in case of a wrong translation term
3. Use `format_date`, which respects the user's timezone and language date format